pyiron_base/project/project_browser.py

```diff
@@ -0,0 +1,337 @@
+                hide_path(bool/None): If True the root_path is omitted in the path.
+        """
+        if Vbox is not None:
+            self.box = Vbox
+        if fix_path is not None:
+            self.fix_path = fix_path
+        if show_files is not None:
+            self._show_files = show_files
+        if hide_path is not None:
+            self._hide_path = hide_path
+        self.refresh()
+
+    def _update_optionbox(self, optionbox):
+
+        def click_option_button(b):
+            if self._busy_check():
+                return
+            self._click_option_button(b)
+            self._busy_check(False)
+
+        set_path_button = widgets.Button(description='Set Path', tooltip="Sets current path to provided string.")
+        set_path_button.on_click(click_option_button)
+        if self.fix_path:
+            set_path_button.disabled = True
+        childs = [set_path_button, self.path_string_box]
+
+        button = widgets.Button(description="Reset selection", width='min-content')
+        button.on_click(click_option_button)
+        childs.append(button)
+
+        optionbox.children = tuple(childs)
+
+    def _click_option_button(self, b):
+        self.output.clear_output(True)
+        with self.output:
+            print('')
+        if b.description == 'Set Path':
+            if self.fix_path:
+                return
+            else:
+                path = self.path
+            if len(self.path_string_box.value) == 0:
+                with self.output:
+                    print('No path given')
+                return
+            elif self.path_string_box.value[0] != '/':
+                path = path + '/' + self.path_string_box.value
+            else:
+                path = self.path_string_box.value
+            self._update_project(path)
+        if b.description == 'Reset selection':
+            self._clickedFiles = []
+            self._data = None
+            self._update_filebox(self.filebox)
+
+    @property
+    def data(self):
+        return self._data
+
+    def _update_project_worker(self, rel_path):
+        try:
+            new_project = self.project[rel_path]
+            # Check if the new_project implements list_nodes()
+            new_project.list_nodes()
+        except (ValueError, AttributeError):
+            self.path_string_box = self.path_string_box.__class__(description="(rel) Path", value='')
+            with self.output:
+                print("No valid path")
+            return
+        else:
+            if new_project is not None:
+                self.project = new_project
+
+    def _update_project(self, path):
+        if isinstance(path, str):
+            rel_path = os.path.relpath(path, self.path)
+            if rel_path == '.':
+                return
+            self._update_project_worker(rel_path)
+        else:
+            self.project = path
+        self.refresh()
+
+    def _update_pathbox(self, box):
```

Comment:
soft recommendation: refactor out a couple chunks into well-named methods to improve readability. This is the first method long/complex enough that I have trouble following it.